src/core/models.py

```diff
@@ -177,12 +179,27 @@ def create(self, **kwargs):
 
 
 class AccountManager(BaseUserManager):
-    def create_user(self, email, password=None, **kwargs):
-        if not email:
-            raise ValueError('Users must have a valid email address.')
+    def create_user(self, username=None, password=None, email=None, **kwargs):
+        """ Creates a user from the given username or email
+        In Janeway, users rely on email addresses to log in. For compatibility
+        with 3rd party libraries, we allow a username argument, however only a
+        email address will be accepted as the username and email.
+        """
+        if not email and username:
+            email = username
+            if "username" in kwargs:
+                del kwargs["username"]
+        try:
+            validate_email(email)
+            self.normalize_email(email)
```

Comment:
Should this be updating the value of `email`? The `normalize_email` method doesn't raise one of the errors you're checking for, it just returns the normalized email:

https://github.com/django/django/blob/1d85b416aa1382e7274e718c79b83dd050fb24ab/django/contrib/auth/base_user.py#L21-L34

Or if you want to keep `email` untransformed, then do we actually need this line?

Reply:
thanks! well spotted